components/sidebar/sidebar_channel/sidebar_base_channel/index.ts

```diff
@@ -3,21 +3,38 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
-import {bindActionCreators, Dispatch} from 'redux';
+import {bindActionCreators, Dispatch, ActionCreatorsMapObject} from 'redux';
 
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
 import {GlobalState} from 'mattermost-redux/types/store';
-import {GenericAction} from 'mattermost-redux/types/actions';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
+import {leaveChannel} from 'actions/views/channel';
 
 import SidebarBaseChannel from './sidebar_base_channel';
 
 function mapStateToProps(state: GlobalState) {
+    const config = getConfig(state);
+
     return {
+        enableXToLeaveChannelsFromLHS: config.EnableXToLeaveChannelsFromLHS,
     };
 }
 
-function mapDispatchToProps(dispatch: Dispatch<GenericAction>) {
+type Actions = {
+    leaveChannel: (channelId: any) => Promise<{
```

Comment:
I suppose there's this:
```
export type ActionResult = {
    data: any;
} | {
    error: any;
};
```

But it doesn't match the actual return type. I'm not going to enforce this level of verbosity around the system, but there's no harm in actually having the actual return type of the function clearly defined, even if it's not used.

Reply:
What's the actual return type? Shouldn't it match be `ActionResult`?

In general, I'd lean towards forcing actions to return a specific type if it would make it easier to use the actions in components.

Reply:
This actually makes it much easier for the component to use the action, since now the component knows it returns the correct state: `{data: any, error: undefined}` or the error state: `{data: undefined, error: any}`

Reply:
Should we change the definition of `ActionResult` then? I'm not understanding why we'd have a font defined and then not use it.

Reply:
It's because `ActionResult` specifies that the `data` field is always an `any`, which isn't a strict type. I think `ActionResult` needs to be a generic like this:

```
export type ActionResult<T> = {
    data: T;
} | {
    error: any;
};
```

Reply:
Will address this as part of: https://mattermost.atlassian.net/browse/MM-22356